pubkey/src/lib.rs

```diff
@@ -163,7 +145,16 @@ impl From<u64> for PubkeyError {
 #[cfg_attr(feature = "bytemuck", derive(Pod, Zeroable))]
 #[derive(Clone, Copy, Default, Eq, Ord, PartialEq, PartialOrd)]
 #[cfg_attr(feature = "dev-context-only-utils", derive(Arbitrary))]
-pub struct Pubkey(pub(crate) [u8; 32]);
+pub struct Pubkey(pub(crate) Address);
+
+impl Deref for Pubkey {
```

Comment:
what's the motivation for this?

Reply:
You could use `Pubkey` references transparently in helpers that expect an `Address`:
```rust
fn expect_address(address: &Address) {
    assert_eq!(address.len(), PUBKEY_BYTES);
}

let address: Address = [1u8; 32];
let pubkey: Pubkey = address.into();

expect_address(&address);
expect_address(&pubkey);
```

Reply:
could also do
```rust
fn expect_address(address: impl AsRef<Address>) {
   let as_ref: &Address = address.as_ref();
    assert_eq!(as_ref.len(), PUBKEY_BYTES);
}
```
This would potentially accept types we don't really want to accept like `Hash`, but it's more obvious behaviour

Reply:
Yeah, that works as well  – I don't have a strong preference. There are a couple of places that we will need to add a type annotation since `Pubkey` already implements `impl AsRef<[u8]> for Pubkey`, but that is not too bad.